test/Integration/SqlOptionsEndToEndTests.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Extensions.Sql.Tests.Common;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql.Tests.Unit
+{
+    public class SqlOptionsEndToEndTests
+    {
+        [Fact]
+        public void ConfigureOptions_AppliesValuesCorrectly_Queues()
+        {
+            string extensionPath = "AzureWebJobs:Extensions:Sql";
+            var values = new System.Collections.Generic.Dictionary<string, string>
+            {
+                { $"{extensionPath}:BatchSize", "30" },
+                { $"{extensionPath}:PollingIntervalMs", "1000" },
+                { $"{extensionPath}:MaxChangesPerWorker", "100" },
+            };
+
+            SqlOptions options = TestHelpers.GetConfiguredOptions<SqlOptions>(b =>
+            {
+                b.AddSql();
+            }, values);
+
+            Assert.Equal(30, options.BatchSize);
```

Comment:
It'd be good to add a test showing that when both host.json and app settings are set it will pick the app settings one.